Sources/SourceKitLSP/SourceKitLSPServer.swift

```diff
@@ -1041,6 +1043,9 @@ extension SourceKitLSPServer {
       codeLensProvider: CodeLensOptions(),
       documentFormattingProvider: .value(DocumentFormattingOptions(workDoneProgress: false)),
       documentRangeFormattingProvider: .value(DocumentRangeFormattingOptions(workDoneProgress: false)),
+      documentOnTypeFormattingProvider: DocumentOnTypeFormattingOptions(triggerCharacters: [
+        "\n", "\r\n", "\r", "{", "}", ";", ".", ":", "#",
+      ]),
```

Comment:
Character list inspired by Xcode.

Reply:
Where did you get the list from in Xcode? Or did you just try different characters?

- `#` seems like an odd choice here. 
- `\r\n` isn’t a character, so not sure if that should be in the list.

Reply:
<img width="825" alt="Screenshot 2024-11-12 at 3 33 27 AM" src="https://github.com/user-attachments/assets/bc6b2d1e-cedc-4a57-bc8e-00c907890987">

> `\r\n` isn’t a character

good catch, i just copy pasted right from Xcode.

Reply:
> # seems like an odd choice here.
Hmm right i did think that as well ... I _could_ remove that.

Reply:
Actually `\r\n` _is_ one `Character`, but 2 unicode scalars.